src/aind_metadata_service/sharepoint/client.py

```diff
@@ -331,32 +506,88 @@ def get_procedure_info(
         ctx = self.client_context
-        list_view = ctx.web.lists.get_by_title(
-            version.value["list_title"]
-        ).views.get_by_title(version.value["view_title"])
-        ctx.load(list_view)
-        ctx.execute_query()
-        list_items = list_view.get_items().filter(filter_string)
-        ctx.load(list_items)
-        ctx.execute_query()
+        for version in ListVersions:
+            filter_string = self._get_filter_string(version, subject_id)
+            list_view = ctx.web.lists.get_by_title(
+                version.value["list_title"]
+            ).views.get_by_title(version.value["view_title"])
+            ctx.load(list_view)
+            ctx.execute_query()
+            list_items = list_view.get_items().filter(filter_string)
+            ctx.load(list_items)
+            ctx.execute_query()
+            (
+                head_frames,
+                craniotomies,
+                fiber_implants,
+                injections,
+            ) = self._map_response(version, list_items)
+            all_head_frames.extend(head_frames)
+            all_craniotomies.extend(craniotomies)
+            all_fiber_implants.extend(fiber_implants)
+            all_injections.extend(injections)
         response = self._handle_response_from_sharepoint(
-            list_items, subject_id=subject_id
+            subject_id=subject_id,
+            head_frames=all_head_frames,
+            craniotomies=all_craniotomies,
+            fiber_implants=all_fiber_implants,
+            injections=all_injections,
         )
-
         return response
 
-    # TODO: Refactor to make less complex?
+    def _map_response(self, version, list_items: ListItemCollection):
+        """
+        Maps sharepoint response to lists of procedures
+        Parameters
+        ----------
+        version : ListVersions
+            Version of the SharePoint List being queried against
+        list_items : ListItemCollection
+            SharePoint returns a ListItemCollection given a query
+        Returns
+        -------
+        head_frames : list
+            Empty list or list of Headframe models
+        craniotomies : list
+            Empty list or list of Craniotomy models
+        fiber_implants : list
+            Empty list or list of FiberImplant models
+        injections : list
+            Empty list or list of BrainInjection models
+        """
+        if version == ListVersions.VERSION_2023:
+            (
+                head_frames,
+                craniotomies,
+                fiber_implants,
+                injections,
+            ) = self._map_2023_response(list_items)
+        else:
+            (
+                head_frames,
+                craniotomies,
+                fiber_implants,
+                injections,
+            ) = self._map_2019_response(list_items)
+        return head_frames, craniotomies, fiber_implants, injections
+
+    @staticmethod
     def _handle_response_from_sharepoint(  # noqa: C901
```

Comment:
It looks like the function is cleaned up. Can you check if the `# noqa: C901` comment is still needed? (flake8 complained the original function was too complex.)